doc/march_packages/march_gait_selection.rst

```diff
@@ -0,0 +1,72 @@
+.. _march-gait-selection-label:
+
+march_gait_selection
+====================
+
+Overview
+--------
+The march_gait_selection module provides the exoskeleton operating system with gait-specific information to perform the
+requested movement. It uses the march-gait-files, which are parsed using the march-shared-classes, to gather this
+information in a standard format. These class objects are stored as attributes in the gait-selection object. The gait
+selection node can use this gait-selection object to extract the data corresponding to the requested gait/subgait name.
+
+Behavior
+--------
+The march-state-machine accepts or declines a gait request originating from the input device. If the gait is accepted
+by the state machine the request will be send over to the march-gait-selection module using the subgait sequence defined
+in the state machine. The gait selection module will extract the data from the parsed gaits, if possible, and if executed
+correctly, will send the required data in a new format to the hardware-interface or simulation part of the exoskeleton
+software.
+
+Transition-subgait
+^^^^^^^^^^^^^^^^^^
+It is possible to transition between two matching named subgaits of two different gaits (for example between the right
+swing of the walk small and walk large). The gait-selection module is able to accomplish this if the requested
+gait/sub-gait format consists of an old and new gait name. This structure requires that both gait names have matching
```

Comment:
Perhaps you could mention the [`GaitName`](https://github.com/project-march/march/blob/develop/march_shared_resources/action/GaitName.action) action here, so that is more clear what the format is.